ssl/ssl_sess.c

```diff
@@ -163,8 +165,14 @@ SSL_SESSION *ssl_session_dup(SSL_SESSION *src, int ticket)
     if (dest->lock == NULL)
         goto err;
 
-    if (src->peer != NULL)
-        X509_up_ref(src->peer);
+    if (!CRYPTO_new_ex_data(CRYPTO_EX_INDEX_SSL_SESSION, dest, &dest->ex_data))
```

Comment:
I've added to the unit tests to demonstrate two different ways of using ex_data, both of which are valid. The new function is generally used to initialize the ex_data structure, so it doesn't necessarily care if it has a usable SSL object yet.